flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -375,7 +387,7 @@ private String operatorName(String suffix) {
       }
 
       // Fallback to use upsert mode parsed from table properties if don't specify in job level.
-      boolean upsertMode = upsert || PropertyUtil.propertyAsBoolean(table.properties(),
+      boolean upsertMode = upsert || PropertyUtil.propertyAsBoolean(props,
```

Comment:
Properties should be passed in here.

Reply:
Are you saying that I should pass the props instead of setting it as a member variable as follow?
https://github.com/apache/iceberg/blob/2ca57336533e1a8a15184006bee77994ca4bf161/flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java#L136

Reply:
I think Ryan meant passing Properties like `distributeDataStream` method

Reply:
ok, updated.